src/main/java/net/minestom/server/inventory/Inventory.java

```diff
@@ -3,392 +3,194 @@
 import net.kyori.adventure.text.Component;
 import net.minestom.server.Viewable;
 import net.minestom.server.entity.Player;
-import net.minestom.server.inventory.click.ClickType;
-import net.minestom.server.inventory.click.InventoryClickResult;
+import net.minestom.server.inventory.click.ClickInfo;
+import net.minestom.server.inventory.click.ClickPreprocessor;
+import net.minestom.server.inventory.click.ClickResult;
 import net.minestom.server.item.ItemStack;
-import net.minestom.server.network.packet.server.play.OpenWindowPacket;
-import net.minestom.server.network.packet.server.play.SetSlotPacket;
-import net.minestom.server.network.packet.server.play.WindowItemsPacket;
-import net.minestom.server.network.packet.server.play.WindowPropertyPacket;
-import net.minestom.server.utils.inventory.PlayerInventoryUtils;
+import net.minestom.server.tag.Taggable;
 import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
 
-import java.util.Collections;
 import java.util.List;
-import java.util.Set;
-import java.util.concurrent.ConcurrentHashMap;
-import java.util.concurrent.CopyOnWriteArraySet;
-import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.UnaryOperator;
 
 /**
- * Represents an inventory which can be viewed by a collection of {@link Player}.
- * <p>
- * You can create one with {@link Inventory#Inventory(InventoryType, String)} or by making your own subclass.
- * It can then be opened using {@link Player#openInventory(Inventory)}.
+ * Represents a generic inventory that can be interacted with.
  */
-public non-sealed class Inventory extends AbstractInventory implements Viewable {
-    private static final AtomicInteger ID_COUNTER = new AtomicInteger();
+public sealed interface Inventory extends Taggable, Viewable permits InventoryImpl {
 
-    // the id of this inventory
-    private final byte id;
-    // the type of this inventory
-    private final InventoryType inventoryType;
-    // the title of this inventory
-    private Component title;
+    interface Typed {
```

Comment:
I believe I am unclear on the use case of `Inventory.Typed`. Why not just put the methods on `ContainerInventory` since this is sealed and only that implements them anyway?

Reply:
the intention was to keep it as clearly an API class (given both Inventory and Inventory.Typed are just interfaces) but that's a fair point. i'll just remove it i guess